aws-janitor/resources/sqs.go

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// SQS queues: https://docs.aws.amazon.com/sdk-for-go/api/service/sqs
+
+type SQSQueues struct{}
+
+func (SQSQueues) MarkAndSweep(opts Options, set *Set) error {
+	logger := logrus.WithField("options", opts)
+	svc := sqs.New(opts.Session, aws.NewConfig().WithRegion(opts.Region))
+
+	input := &sqs.ListQueuesInput{}
+
+	var toDelete []*sqsQueue // Paged call, defer deletion until we have the whole list.
+
+	pageFunc := func(page *sqs.ListQueuesOutput, _ bool) bool {
+		for _, url := range page.QueueUrls {
+			attrInput := &sqs.GetQueueAttributesInput{
+				AttributeNames: []*string{aws.String(sqs.QueueAttributeNameAll)},
+				QueueUrl:       url,
+			}
+			attr, err := svc.GetQueueAttributes(attrInput)
+			if err != nil {
+				return false
+			}
+
+			q := &sqsQueue{
+				Account:  opts.Account,
+				Region:   opts.Region,
+				Name:     *attr.Attributes[sqs.QueueAttributeNameQueueArn],
+				QueueURL: *url,
+			}
+			svcRules := eventbridge.New(opts.Session, aws.NewConfig().WithRegion(opts.Region))
+
+			// Only delete rules that uses SQS queue as target. There are default rules that should not be deleted.
+			rules, err := svcRules.ListRuleNamesByTarget(&eventbridge.ListRuleNamesByTargetInput{
+				TargetArn: attr.Attributes[sqs.QueueAttributeNameQueueArn],
+			})
+			if err != nil {
+				logger.Warningf("listing rules by target failed: %s", err.Error())
+			}
+
+			for _, rule := range rules.RuleNames {
+				deleteEventBridgeRule(rule, svcRules, logger)
+			}
+
+			unitTimestamp, _ := strconv.ParseInt(*attr.Attributes[sqs.QueueAttributeNameCreatedTimestamp], 10, 64)
```

Comment:
I think this might want to be `aws.StringValue` instead of simply dereferencing, since if `QueueAttributeNameCreatedTimestamp` is missing for some reason, this would be a nil dereference.